src/lang_items.rs

```diff
@@ -48,25 +55,33 @@ fn panic_handler(_info: &PanicInfo) -> ! {
     super::debug::low_level_status_code(1);
 
     // Flash all LEDs (if available).
-    loop {
-        for led in led::all() {
-            led.on();
-        }
-        timer::sleep(Duration::from_ms(100));
-        for led in led::all() {
-            led.off();
+    executor::block_on(async {
```

Comment:
It seems unnecessary for this to build a future if it will loop infinitely. Why not have the `loop {}` on the outside and call `executor::block_in` on the sleep calls inside the loop?

I have the same concern about cycle_leds() below.

Reply:
Why is it unnecessary? It makes the code read nicer than having a future for each call of `sleep`.

I have no strong feelings for this code. It should probably get removed and be replaced by something else. Maybe it should print to the console and restart the application?

Reply:
The low_level_status_code(1) call above will trigger a print if the LowLevelDebug driver is present; that's probably the best option. I don't think it should restart the application automatically by default, although I could see that being a nice option.

Reply:
I think a panic! should be visible even if no console is attached if the device is running somewhere "in production" so I like the solution with the blinking leds.